api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -2512,6 +2508,63 @@ public void testShareWorkspaceRemoveBillingProjectUser() {
     verify(fireCloudService, times(1)).addOwnerToBillingProject(any(), any());
   }
 
+  @Test
+  public void testShareWorkspacePublishedWorkspace() {
+    DbCdrVersion ctCdrVersion =
+        TestMockFactory.createControlledTierCdrVersion(cdrVersionDao, accessTierDao, 5L);
+
+    DbUser writerUser = createAndSaveUser("writerfriend@gmail.com", 124L);
+
+    when(fireCloudService.getWorkspaceAclAsService(anyString(), anyString()))
+        .thenReturn(
+            createWorkspaceACL(
+                new JSONObject()
+                    .put(
+                        // Specifically, the REGISTERED tier is used for publishing.
+                        accessTier.getAuthDomainGroupEmail(),
```

Comment:
could also rename this to `registeredTier`

Reply:
Good point, done